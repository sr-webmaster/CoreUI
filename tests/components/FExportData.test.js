import { mount, createLocalVue } from '@vue/test-utils'
import Component from '../../src/components/FExportData'

describe('Export Data', () => {
  let localVue

  beforeEach(() => {
    localVue = createLocalVue()
  })

  describe('Methods', () => {
    test('export emits export event with payload', () => {
      const wrapper = mount(Component, { localVue })

      wrapper.setData({
        opt: 'pdf'
      })

      wrapper.vm.exportData()
      expect(wrapper.emitted().export).toHaveLength(1)
      expect(wrapper.emitted().export[0]).toEqual([{
        opt: 'pdf'
      }])
    })
  })

  describe('Visuals', () => {
    test('Props option', () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          option: 'pdf'
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })

    test('click on export button', async () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          option: 'pdf'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
      wrapper.find('button').trigger('click.prevent')
      await wrapper.vm.$nextTick()
      // Test radio button
      wrapper.find('input[type="radio"][value="csv"]').element.selected = true
      wrapper.find('input[type="radio"][value="csv"]').trigger('change')
      wrapper.findAll('button').trigger('click')
      expect(wrapper.emitted('close')).toHaveLength(2)
      expect(wrapper.emitted('export')).toHaveLength(1)
      expect(wrapper.emitted().export[0]).toEqual([{
        opt: 'csv'
      }])
    })
  })
})
