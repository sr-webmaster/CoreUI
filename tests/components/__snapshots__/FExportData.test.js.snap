// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export Data Visuals Props option 1`] = `
<div
  class="f-card-export v-card v-sheet theme--light"
>
  <!---->
   
  <!---->
   
  <div
    class="v-card__text"
  >
    <form
      class="v-form"
      novalidate="novalidate"
    >
      <div
        class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
            style="height: auto;"
          >
            <div
              class="v-input--radio-group__input"
              role="radiogroup"
            >
              <div
                class="v-radio theme--light"
              >
                <div
                  class="v-input--selection-controls__input"
                >
                  <input
                    aria-checked="false"
                    aria-label="PDF"
                    name="v-radio-22"
                    role="radio"
                    type="radio"
                    value="pdf"
                  />
                  <div
                    class="v-input--selection-controls__ripple"
                  />
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    radio_button_unchecked
                  </i>
                </div>
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: relative;"
                >
                  PDF
                </label>
              </div>
               
              <div
                class="v-radio theme--light"
              >
                <div
                  class="v-input--selection-controls__input"
                >
                  <input
                    aria-checked="false"
                    aria-label="CSV"
                    name="v-radio-22"
                    role="radio"
                    type="radio"
                    value="csv"
                  />
                  <div
                    class="v-input--selection-controls__ripple"
                  />
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    radio_button_unchecked
                  </i>
                </div>
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: relative;"
                >
                  CSV
                </label>
              </div>
            </div>
          </div>
          <div
            class="v-messages theme--light"
          >
            <div
              class="v-messages__wrapper"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
   
  <div
    class="v-card__actions"
  >
    <button
      class="v-btn v-btn--block v-btn--flat theme--light secondary--text"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Cancel
    
      </div>
    </button>
     
    <button
      class="v-btn v-btn--block theme--light primary"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Export
    
      </div>
    </button>
  </div>
</div>
`;

exports[`Export Data Visuals click on export button 1`] = `
<div
  class="f-card-export v-card v-sheet theme--light"
>
  <!---->
   
  <!---->
   
  <div
    class="v-card__text"
  >
    <form
      class="v-form"
      novalidate="novalidate"
    >
      <div
        class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
            style="height: auto;"
          >
            <div
              class="v-input--radio-group__input"
              role="radiogroup"
            >
              <div
                class="v-radio theme--light"
              >
                <div
                  class="v-input--selection-controls__input"
                >
                  <input
                    aria-checked="false"
                    aria-label="PDF"
                    name="v-radio-35"
                    role="radio"
                    type="radio"
                    value="pdf"
                  />
                  <div
                    class="v-input--selection-controls__ripple"
                  />
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    radio_button_unchecked
                  </i>
                </div>
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: relative;"
                >
                  PDF
                </label>
              </div>
               
              <div
                class="v-radio theme--light"
              >
                <div
                  class="v-input--selection-controls__input"
                >
                  <input
                    aria-checked="false"
                    aria-label="CSV"
                    name="v-radio-35"
                    role="radio"
                    type="radio"
                    value="csv"
                  />
                  <div
                    class="v-input--selection-controls__ripple"
                  />
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    radio_button_unchecked
                  </i>
                </div>
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: relative;"
                >
                  CSV
                </label>
              </div>
            </div>
          </div>
          <div
            class="v-messages theme--light"
          >
            <div
              class="v-messages__wrapper"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
   
  <div
    class="v-card__actions"
  >
    <button
      class="v-btn v-btn--block v-btn--flat theme--light secondary--text"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Cancel
    
      </div>
    </button>
     
    <button
      class="v-btn v-btn--block theme--light primary"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Export
    
      </div>
    </button>
  </div>
</div>
`;
